src/ansiblelint/config.py

```diff
@@ -171,3 +182,75 @@ def parse_ansible_version(stdout: str) -> tuple[str, str | None]:
     if match:
         return match.group(1), None
     return "", f"FATAL: Unable parse ansible cli version: {stdout}"
+
+
+def in_venv() -> bool:
+    """Determine whether Python is running from a venv."""
+    if hasattr(sys, "real_prefix"):
+        return True
+    pfx = getattr(sys, "base_prefix", sys.prefix)
+    return pfx != sys.prefix
+
+
+def guess_install_method() -> str:
+    """Guess if upgrade command installed."""
+    pip = ""
+    if in_venv():
+        # for k in ("VIRTUAL_ENV", "CONDA_PREFIX", "PYENV_VERSION"):
+        #     if k in os.environ:
+        _logger.debug("Found virtualenv, assuming `pip3 install` will work.")
+        pip = f"pip install --upgrade {__package__}"
+    elif __file__.startswith(os.path.expanduser("~/.local/lib")):
+        _logger.debug(
+            "Found --user installation, assuming `pip3 install --user` will work."
+        )
+        pip = f"pip3 install --user --upgrade {__package__}"
+
+    return pip
+
+
+def get_version_warning() -> str:
+    """Display warning if current version is outdated."""
+    msg = ""
+    new_version = Version("9999.99.9")
+    data = {}
+    current_version = Version(__version__)
+    cache_dir = os.path.expanduser("~/.cache/ansible-lint")
```

Comment:
Can't we use ``XDG_`` stuff ? It's already used in ``yaml_utils.py`` after all

Reply:
Indeed, already fixed.